src/prover/poseidon_hasher.rs

```diff
@@ -0,0 +1,31 @@
+use poseidon_rs::{Fr, Poseidon};
+
+use crate::types::types::{ToVecFr, ToU32};
```

Comment:
What do you think about this style?

```suggestion
use poseidon_rs::Fr;
use poseidon_rs::Poseidon;

use crate::types::types::ToU32;
use crate::types::types::ToVecFr;
```

I generally prefer it because there's less line rewrapping as the import lists inevitably get too long.